MathLib/LinAlg/BLAS.cpp

```diff
@@ -0,0 +1,204 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "BLAS.h"
+
+
+// Global PETScMatrix/PETScVector //////////////////////////////////////////
+#ifdef USE_PETSC
+
+#include "MathLib/LinAlg/PETSc/PETScVector.h"
+#include "MathLib/LinAlg/PETSc/PETScMatrix.h"
+
+namespace MathLib { namespace BLAS
+{
+
+// Vector
+
+void copy(PETScVector const& x, PETScVector& y)
+{
+    y = x;
+}
+
+void scale(PETScVector& x, double const a)
+{
+    VecScale(x.getRawVector(), a);
+}
+
+// y = a*y + X
+void aypx(PETScVector& y, double const a, PETScVector const& x)
+{
+    // TODO check sizes
+    VecAYPX(y.getRawVector(), a, x.getRawVector());
+}
+
+// y = a*x + y
+void axpy(PETScVector& y, double const a, PETScVector const& x)
+{
+    // TODO check sizes
+    VecAXPY(y.getRawVector(), a, x.getRawVector());
+}
+
+// y = a*x + y
+void axpby(PETScVector& y, double const a, double const b, PETScVector const& x)
+{
+    // TODO check sizes
+    VecAXPBY(y.getRawVector(), a, b, x.getRawVector());
+}
+
+
```

Comment:
Nevertheless I'd like to keep the two empty lines as they are, because the Matrix section starts here.